dico_toolbox/graph.py

```
@@ -62,27 +92,135 @@ def stack_buckets(vertex, labels=["aims_ss", "aims_bottom", "aims_other"]):
+            When return_keys is not None, if a key is not defined in a vertex, 
+            defaults are used to set up the value. If defaults is a single value,
+            the value is used for all the keys. Default is None.
+
+        transform: "ICBM2009c" | "Talairach" | None (opt.)
+            Apply known transformation. Two spaces are available: "Talairach" 
+            and "ICBM2009c"(refers to the MNI space used by BrainVISA).
+            None value skip the transformation step.
+            Default is None.
+
+        bck_types: list of str (opt.)
+            Bucket keys used to list points. 
+            Default is ['aims_ss', 'aims_other', 'aims_bottom'].
+
+        Return
+        ======
+        stacked buckets and values dict (or list if return_keys is a string) if asked.
+
+
+        Examples
+        ========
+        Return buckets of all vertex
+        '''python
+        stack_buckets(graph, 'name', return_keys='name', default='unknown', transform="Talairach")
+        '''
+
+        Return buckets, name and vertex index of automatically labeled as 'S.T.S.asc' in the image space. 
+        Name and vertex index is set to None if not defined.
+        '''python
+        stack_buckets(graph, 'label', needed=['S.T.S.asc'], return_keys=['name', 'vertex_index'], default=None)
+        '''
+
+        Return buckets and names of all vertex (with name set to 'unknown' if undefined)
+        '''python
+        stack_buckets(graph, 'name', return_keys='name', default='unknown', transform="Talairach")
+        '''
+
     """
-    graph_buckets = list()
-    tr_aims = _aims.GraphManip.getICBM2009cTemplateTransform(graph)
+    if isinstance(graph, dict):
+        # It is actually a vertex (because of previous implementation)
+        _warnings.warn("stack_buckets() is now used for graph. Use stack_vertex_buckets instead()",
+                       _warnings.DeprecationWarning)
+        return stack_vertex_buckets(graph, bck_types=BUCKETS_TYPES)
```

raise instead of warning here